src/main/java/org/spongepowered/api/event/cause/EventContextKeys.java

```diff
@@ -45,7 +45,7 @@
  */
 public final class EventContextKeys {
 
-    public static final EventContextKey<UUID> CREATOR = createFor("CREATOR");
+    public static final EventContextKey<User> CREATOR = createFor("CREATOR");
```

Comment:
Or not even turn it into a user, because they are storing references to UUID's ?

I'll defer to what you end up finding though as long as this change gets a second look, you are more experienced then I.